Soundify/Soundify/Scenes/Track/TrackDetailViewController.swift

```diff
@@ -0,0 +1,165 @@
+    }
+    
+    private func configImageView() {
+        trackImageView.do {
+            $0.sd_setImage(with: track.album?.images.urlImage, completed: nil)
+            $0.borderColor = .lightGray
+            $0.borderWidth = 1
+            $0.cornerRadius = 10
+        }
+        
+        backgroundImageView.do {
+            $0.sd_setImage(with: track.album?.images.urlImage, completed: nil)
+            $0.alpha = 0.1
+        }
+    }
+    
+    private func getAudio() {
+        guard let url = URL(string: track.previewUrl ?? "") else {
+            artistsNameLabel.do {
+                $0.numberOfLines = 0
+                if let text = $0.text { $0.text = "\(text)\n No preview Url" }
+            }
+            return
+        }
+        
+        
+        player = AVPlayer(playerItem: AVPlayerItem(url: url) ).then {
+            $0.play()
+            let interval = CMTime(value: 1, timescale: 1000)
+            $0.addPeriodicTimeObserver(forInterval: interval, queue: .main) { [weak self] (currentTime) in
+                if self?.totalTime.isNaN ?? true { return }
+                self?.configLabelTime(at: currentTime)
+                self?.playbackWhenFinishSong()
+            }
+        }
+        
+        playSongWhenEnterBackground()
+    }
+    
+    private func configLabelTime(at currentTime: CMTime) {
+        audioSlider.value = Float(currentTime.seconds) / (totalTime)
+        totalTimeLabel.text = Double(totalTime).formatterMinuteAndSecondInTrack
+        currentTimeLabel.text = currentTime.seconds.formatterMinuteAndSecondInTrack
+    }
+    
+    private func playbackWhenFinishSong() {
+        if currentTimeLabel.text == totalTimeLabel.text {
+            player?.seek(to: CMTime(seconds: 0, preferredTimescale: 1000))
+            player?.play()
+        }
+    }
+    
+    private func playSongWhenEnterBackground() {
+        do {
+            try AVAudioSession.sharedInstance().setCategory(.playback)
+        } catch {
+            print(error)
+        }
+    }
+}
+
+//MARK: - IBAction
+extension TrackDetailViewController {
+    
+    @IBAction func sliderValueChanged(_ sender: UISlider, forEvent event: UIEvent) {
+        if let touchEvent = event.allTouches?.first {
+            let currentTime = Double(sender.value * self.totalTime)
+            isPlaying = false
+            switch touchEvent.phase {
+            case .moved:
+                DispatchQueue.main.async {
+                    self.currentTimeLabel.text = currentTime.formatterMinuteAndSecondInTrack
+                }
+                break
+                
+            case .ended:
+                player?.seek(to: CMTime(seconds: currentTime, preferredTimescale: 1000))
+                isPlaying = true
+                break
+                
+            default:
+                break
+            }
+        }
+    }
+    
+    @IBAction func playButtonClicked(_ sender: UIButton) {
+        if isPlaying {
```

Comment:
```suggestion
       isPlaying = !isPlaying
```

Reply:
:v Long gà vãi, dùng như này đỉnh hơn
```suggestion
        isPlaying.toggle()
```